### DISCUSSION TITLE
Need help with curl

### DISCUSSION THREAD

**Discussion by KaniketCode95** [curl/curl#13876] [Q&A]
Hi,
I have a camera device so through its terminal i want to post logs to a hosted server.
What i did here -
1. created a local host server with help of node
2. verified with postman api
3. And now im trying to send data to the local host through curl command to
the server but its not working.

And I have tried same command in windows CMD and Virtualbox terminal 
and its working 

command : 
1. curl -X POST http://172.30.53.19:8000/
2. curl -X POST --data-binary @kk.json http://172.30.53.19:8000/

what could be the reason/solution

**Comment by jzakrzewski** [comment]
1. Drop `-X POST`
2. add `-v` to get some details of whats going on

**Comment by KaniketCode95** [comment]
![Screenshot 2024-06-04 170528](https://github.com/curl/curl/assets/72891979/3ddeefd9-ec98-494a-bf9f-95ce3efc00b6)
![Screenshot 2024-06-04 170551](https://github.com/curl/curl/assets/72891979/886577dc-0962-4950-87d8-a2deef2a1415)

**Reply by KaniketCode95** [reply]
1st one is my device terminal and second one is wsl.


virtualbox result:
~~~
*   Trying 172.30.53.19:8000...
* TCP_NODELAY set
* Connected to 172.30.53.19 (172.30.53.19) port 8000 (#0)
> POST / HTTP/1.1
> Host: 172.30.53.19:8000
> User-Agent: curl/7.68.0
> Accept: */*
> Content-Length: 6315
> Content-Type: application/x-www-form-urlencoded
> Expect: 100-continue
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 100 Continue
* We are completely uploaded and fine
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Date: Tue, 04 Jun 2024 11:33:36 GMT
< Connection: keep-alive
< Keep-Alive: timeout=5
< Content-Length: 12
< 
* Connection #0 to host 172.30.53.19 left intact
~~~

**Reply by bagder** [reply]
(First: please show text as text, not images, since it will be easier to read and allows searches properly etc)

In that first case where curl says nothing at all (I guess that is what you are asking about?), even with `-v`, are you sure that is actually running curl? If it literally says nothing at all even with `-v`, it seems strange.

Which curl does it run ( `which curl`)? What if you just run `curl -V` there, does that work?

**Reply by KaniketCode95** [reply]
Yes, looking for first case.
here is the curl version 



# curl --version
curl 7.81.0 (arm-buildroot-linux-gnueabihf) libcurl/7.81.0 OpenSSL/1.1.1m zlib/1.2.11 libidn2/2.3.2
Release-Date: 2022-01-05
Protocols: dict file ftp ftps gopher gophers http https imap imaps mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS HSTS HTTPS-proxy IDN IPv6 Largefile libz NTLM SSL TLS-SRP UnixSockets
#

**Reply by jay** [reply]
I'm guessing this is some sort of IO problem that has nothing to do with curl. You can try `strace curl ...` to see if it's actually hanging

**Reply by KaniketCode95** [reply]
Hi,
I have got the solution, it was something has to done with network not with the curl.
I have created another server with xampp and able to solve the problem.

Thanks for the immediate responses of yours 
❤️❤️
